modules/config/ConfigDecoders.scala

```diff
@@ -0,0 +1,122 @@
+    */
+  object std {
+    // format: OFF
+    def config        (path: String): ConfigDecoder[Config]         = instance(path)(_ getConfig _)
+    def string        (path: String): ConfigDecoder[String]         = instance(path)(_ getString _)
+    def number        (path: String): ConfigDecoder[Number]         = instance(path)(_ getNumber _)      //#=typesafe
+    def boolean       (path: String): ConfigDecoder[Boolean]        = instance(path)(_ getBoolean _)
+    def int           (path: String): ConfigDecoder[Int]            = instance(path)(_ getInt _)
+    def long          (path: String): ConfigDecoder[Long]           = instance(path)(_ getLong _)
+    def double        (path: String): ConfigDecoder[Double]         = instance(path)(_ getDouble _)
+    def finiteDuration(path: String): ConfigDecoder[FiniteDuration] = instance(path)(_ getDuration _)
+
+    def configList    (path: String): ConfigDecoder[List[Config]]   = instance(path)(_ getConfigList _)
+    def stringList    (path: String): ConfigDecoder[List[String]]   = instance(path)(_ getStringList _)
+    def numberList    (path: String): ConfigDecoder[List[Number]]   = instance(path)(_ getNumberList _)  //#=typesafe
+    def booleanList   (path: String): ConfigDecoder[List[Boolean]]  = instance(path)(_ getBooleanList _)
+    def intList       (path: String): ConfigDecoder[List[Int]]      = instance(path)(_ getIntList _)
+    def longList      (path: String): ConfigDecoder[List[Long]]     = instance(path)(_ getLongList _)
+    def doubleList    (path: String): ConfigDecoder[List[Double]]   = instance(path)(_ getDoubleList _)
+    // format: ON
+
+    @inline private[this] def instance[A: ClassTag](
+      path: String)(f: (Config, String) => A): ConfigDecoder[A] =
+      Decoder.instance(config =>
+        try {
+          f(config, path).right
+        } catch {
+          case e: ConfigException.Missing   => Missing.atPath(path).left
+          case e: ConfigException.WrongType => WrongType(classTag[A].toString).atPath(path).left //#=typesafe
+          //#+shocon
+          case e: MatchError if e.getMessage == "null" =>
+            Missing.atPath(path).left
+          //#-shocon
+          case other: Throwable             => Underlying(other).atPath(path).left
+        }
+      )
+
+    @inline private[this] implicit def convertLists[A, B](j: java.util.List[A]): List[B] =
+      j.asScala.toList.map(_.asInstanceOf[B])
+
+    @inline private[this] implicit def convertDuration(j: java.time.Duration): FiniteDuration =
+      scala.concurrent.duration.FiniteDuration(j.toNanos, java.util.concurrent.TimeUnit.NANOSECONDS)
+  }
+
+  /** Typesafe config decoders that prevent narrowing of configuration
+    * data during the loading process
+    */
+  object strict {
+    // coming soon...
+  }
+
+
+  /** The backend implementation. Either "typesafe" or "shocon" */ //#=typesafe
+  final val BACKEND: String = "typesafe"                           //#=typesafe
```

Comment:
We should consider what would be the implications of supporting something like Consul down the road. Most industry deployments are based on that and expect hot reloadable config. We are also going to need that for Freestyle eventually.